test/xbgpu/test_engine.py

```diff
@@ -590,30 +697,34 @@ async def test_saturation(
         n_baselines = n_ants * (n_ants + 1) * 2
 
         def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
-            timestamp = timestamp_step * batch_index
+            timestamp = batch_index * timestamp_step
```

Comment:
Indeed indeed, mostly because it bugged me that the `heap_factory` in `test_xengine_end_to_end` (~Line 570) had this timestamp calculation as
> `timestamp = batch_index * timestamp_step`

And this was the other way around (for no apparent reason?).

Reply:
That's a good reason to swap it around then.